samples/extensions/sparse_image/sparse_image.cpp

```diff
@@ -0,0 +1,1512 @@
+	VkPipelineDynamicStateCreateInfo dynamic = vkb::initializers::pipeline_dynamic_state_create_info(dynamics.data(), vkb::to_u32(dynamics.size()));
+
+	// Load our SPIR-V shaders.
+	std::array<VkPipelineShaderStageCreateInfo, 2U> shader_stages{};
+
+	// Vertex stage of the pipeline
+	shader_stages[0] = load_shader("sparse_image/sparse.vert", VK_SHADER_STAGE_VERTEX_BIT);
+	shader_stages[1] = load_shader("sparse_image/sparse.frag", VK_SHADER_STAGE_FRAGMENT_BIT);
+
+	// We need to specify the pipeline layout and the render pass description up front as well.
+	VkGraphicsPipelineCreateInfo pipeline_create_info = vkb::initializers::pipeline_create_info(sample_pipeline_layout, render_pass);
+	pipeline_create_info.stageCount                   = vkb::to_u32(shader_stages.size());
+	pipeline_create_info.pStages                      = shader_stages.data();
+	pipeline_create_info.pVertexInputState            = &vertex_input;
+	pipeline_create_info.pInputAssemblyState          = &input_assembly;
+	pipeline_create_info.pRasterizationState          = &raster;
+	pipeline_create_info.pColorBlendState             = &blend;
+	pipeline_create_info.pMultisampleState            = &multisample;
+	pipeline_create_info.pViewportState               = &viewport;
+	pipeline_create_info.pDepthStencilState           = &depth_stencil;
+	pipeline_create_info.pDynamicState                = &dynamic;
+
+	VK_CHECK(vkCreateGraphicsPipelines(get_device().get_handle(), pipeline_cache, 1U, &pipeline_create_info, nullptr, &sample_pipeline));
+}
+
+void SparseImage::setup_camera()
+{
+	camera.type = vkb::CameraType::FirstPerson;
+	camera.set_perspective(fov_degrees, static_cast<float>(width) / static_cast<float>(height), 0.1f, 512.0f);
+	camera.set_rotation(glm::vec3(0.0f, 0.0f, 0.0f));
+	camera.set_translation(glm::vec3(0.0f, 0.0f, -50.0f));
+	camera.translation_speed = 3.0f;
+}
+
+/**
+ * 	@brief Copy a single-page, raw-pixel-data from the CPU memory into the stagging buffer.
+ */
+void SparseImage::copy_single_raw_data_block(uint8_t buffer[], const VkExtent2D blockDim, VkOffset2D offset, size_t stride)
+{
+	for (size_t row = 0U; row < blockDim.height; row++)
+	{
+		size_t position = (row + offset.y) * stride + offset.x * 4U;
+		memcpy(&buffer[row * blockDim.width * 4U], &virtual_texture.raw_data_image->get_data()[position], blockDim.width * 4U);
+	}
+}
+
+/**
+ * 	@brief Fill up the information on how the sparse image should be bound and call vkQueueBindSparse, based on the bind_list.
+ */
+void SparseImage::bind_sparse_image()
+{
+	virtual_texture.bind_list.sort();
+
+	for (size_t count = 0U; count < virtual_texture.num_pages; count++)
+	{
+		if (virtual_texture.bind_list.empty() == true)
+		{
+			virtual_texture.sparse_image_memory_bind[count].memory = VK_NULL_HANDLE;
+			continue;
+		}
+
+		size_t page_index = virtual_texture.bind_list.front();
+		if (count != page_index)
+		{
+			virtual_texture.sparse_image_memory_bind[count].memory = VK_NULL_HANDLE;
+			continue;
+		}
```

Comment:
You don't need this page_index anymore (as it equals count from here on).
That is, maybe just check
`if (virtual_texture.bind_list.empty() || ( virtual_texture.bind_list.front() != count))`